src/sources/fluent/mod.rs

```diff
@@ -340,12 +377,37 @@ impl Decoder for FluentEntryStreamDecoder {
     }
 }
 
+struct FluentAcker {
+    chunk: Option<String>,
+}
+
+impl FluentAcker {
+    fn new(frame: &FluentFrame) -> Self {
+        Self {
+            chunk: frame.chunk.clone(),
+        }
+    }
+}
+
+impl TcpSourceAcker for FluentAcker {
+    fn build_ack(self, ack: TcpSourceAck) -> Bytes {
+        match self.chunk {
+            None => Bytes::new(),
+            Some(chunk) => match ack {
+                TcpSourceAck::Ack => format!(r#"{{"ack": "{}"}}"#, chunk).into(),
+                _ => "{}".into(),
```

Comment:
Is it actually supposed to send back an empty JSON object if there was an error/rejection?

Reply:
Yes, weird as that may seem. https://github.com/fluent/fluentd/wiki/Forward-Protocol-Specification-v0#response